### ISSUE TITLE
Network Interface not detecting removal of public IP

### ISSUE THREAD

**Issue by cehoffman** [terraform-providers/terraform-provider-azurerm#564]
### Terraform Version
v0.10.8

### Affected Resource(s)

- azurerm_virtual_network_interface

### Terraform Configuration Files
```hcl
resource "azurerm_network_interface" "nic1" {
  name                = "nic1"
  location            = "${var.location}"
  resource_group_name = "${var.rsg}"

  ip_configuration {
    name                          = "ipconfig"
    subnet_id                     = "${azurerm_subnet.0_25.id}"
    private_ip_address_allocation = "dynamic"
    public_ip_address_id          = "${azurerm_public_ip.vm.id}"
  }
}

resource "azurerm_public_ip" "vm" {
  name                         = "vm1-ip"
  resource_group_name          = "${var.rsg}"
  location                     = "${var.location}"
  public_ip_address_allocation = "static"
}

resource "azurerm_virtual_network" "vpn" {
  name                = "VPN"
  location            = "${var.location}"
  resource_group_name = "${var.rsg}"
  address_space       = ["10.38.200.0/24"]
}

resource "azurerm_subnet" "0_25" {
  name                 = "10.38.200.0-25"
  resource_group_name  = "${var.rsg}"
  virtual_network_name = "${azurerm_virtual_network.vpn.name}"
  address_prefix       = "10.38.200.0/25"
}
```

### Debug Output
https://gist.github.com/cehoffman/f71e10b952824c659b38a060ae637a65

### Expected Behavior

It should remove the public ip resource from the nic.

### Actual Behavior

No changes are planned.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `Remove public ip id from nic`
2. `terraform plan`

**Comment by mocofound** [comment]
I'm not sure if this is a bug or enhancement request, but I was able to reproduce this as described.

**Comment by tombuildsstuff** [comment]
hi @cehoffman @mocofound 

Thanks for opening this issue :)

I've taken a look into this and this is because the `public_ip_address_id` field within the `azurerm_network_interface` resource is `Computed`. We'll need to add a test to confirm this, but we should be able to fix this by remove the `Computed` flag on the `public_ip_address_id` field. In  the interim you can work around this by setting this field to an empty string.

Thanks!

**Comment by JunyiYi** [comment]
hi @cehoffman @mocofound , this issue has been fixed in [v1.7.0](https://github.com/terraform-providers/terraform-provider-azurerm/blob/bce918a9702a7d700e47b52577f4727bfb0a6e90/CHANGELOG.md#170-june-16-2018).
